geopandas/geodataframe.py

```diff
@@ -1552,35 +1566,70 @@ def explode(self, column=None, **kwargs):
         0  name1  MULTIPOINT (1.00000 2.00000, 3.00000 4.00000)
         1  name2  MULTIPOINT (2.00000 1.00000, 0.00000 0.00000)
 
-        >>> exploded = gdf.explode()
+        >>> exploded = gdf.explode(add_multiindex=True)
         >>> exploded
               col1                 geometry
         0 0  name1  POINT (1.00000 2.00000)
           1  name1  POINT (3.00000 4.00000)
         1 0  name2  POINT (2.00000 1.00000)
           1  name2  POINT (0.00000 0.00000)
 
+        >>> exploded = gdf.explode(add_multiindex=False)
+        >>> exploded
+            col1                 geometry
+        0  name1  POINT (1.00000 2.00000)
+        0  name1  POINT (3.00000 4.00000)
+        1  name2  POINT (2.00000 1.00000)
+        1  name2  POINT (0.00000 0.00000)
+
+        >>> exploded = gdf.explode(ignore_index=True)
+        >>> exploded
+            col1                 geometry
+        0  name1  POINT (1.00000 2.00000)
+        1  name1  POINT (3.00000 4.00000)
+        2  name2  POINT (2.00000 1.00000)
+        3  name2  POINT (0.00000 0.00000)
+
+
         See also
         --------
         GeoDataFrame.dissolve : dissolve geometries into a single observation.
 
         """
 
+        warnings.warn(
+            "add_multiindex defaults to True to be consistent with old behaviour. "
+            "In GeoPandas 1.0 will default to False to be consistent with Pandas.",
+            FutureWarning,
+            stacklevel=2,
+        )
```

Comment:
```suggestion
        if add_multiindex is None:
            warnings.warn(
                "add_multiindex defaults to True to be consistent with old behaviour. "
                "In GeoPandas 1.0 will default to False to be consistent with Pandas. "
                "Use add_multiindex=True to keep the current behaviour and silent "
                "the warning.",
                FutureWarning,
                stacklevel=2,
            )
            add_multiindex = True
```

As per the comment above. You may need to reformat it.